packages/ui/src/components/Form/customField/CustomNestField.tsx

```diff
@@ -0,0 +1,91 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Button, Card, CardBody, CardExpandableContent, CardHeader, CardTitle } from '@patternfly/react-core';
+import { useMemo, useState } from 'react';
+import { HTMLFieldProps, connectField, filterDOMProps } from 'uniforms';
+import { isDefined } from '../../../utils';
+import { CustomAutoField } from '../CustomAutoField';
+import './CustomNestField.scss';
+
+export type CustomNestFieldProps = HTMLFieldProps<
+  object,
+  HTMLDivElement,
+  { properties?: Record<string, unknown>; helperText?: string; itemProps?: object }
+>;
+
+export const CustomNestField = connectField(
+  ({
+    children,
+    error,
+    errorMessage,
+    fields,
+    itemProps,
+    label,
+    name,
+    showInlineError,
+    disabled,
+    ...props
+  }: CustomNestFieldProps) => {
+    const [isExpanded, setIsExpanded] = useState(false);
+
+    const advancedFields = useMemo(() => {
+      const advanced: string[] = [];
+      if (isDefined(props.properties)) {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        Object.entries(props.properties as Record<string, any>).forEach(([key, value]) => {
+          if (isDefined(value.labels) && value.labels.includes('advanced')) {
+            advanced.push(key);
+          }
+        });
+      }
+      return advanced;
+    }, [props.properties]);
+
+    const nodes =
+      children ||
+      fields
+        ?.filter((field) => !advancedFields.includes(field))
+        .map((field) => <CustomAutoField key={field} name={field} />);
+    const advancedNodes = advancedFields?.map((field) => <CustomAutoField key={field} name={field} />);
+
+    return (
+      <Card data-testid={'nest-field'} {...filterDOMProps(props)} isExpanded={isExpanded}>
+        <CardHeader>
+          <CardTitle>{label}</CardTitle>
+        </CardHeader>
+        <CardBody className="custom-nestfield-spacing">{nodes}</CardBody>
+        {advancedNodes.length > 0 && (
+          <>
+            <CardHeader onExpand={() => setIsExpanded(!isExpanded)}>
+              <CardTitle>
+                <Button variant="link" isInline onClick={() => setIsExpanded(!isExpanded)}>
+                  Advanced properties
+                </Button>
+              </CardTitle>
+            </CardHeader>
+            <CardExpandableContent>
+              <CardBody className="custom-nestfield-spacing">{advancedNodes}</CardBody>
+            </CardExpandableContent>
+          </>
+        )}
+      </Card>
+    );
+  },
+);
```

Comment:
After checking it, I think it's a good approach, but there's still an open question for the future, not exactly related to this PR which is "how to identify components in consumer or producer mode", so we can filter such properties.

I tinkered a bit with your idea and I have a proposal, please let me know what you think:
```tsx
import { Card, CardBody, CardHeader, CardTitle, ExpandableSection } from '@patternfly/react-core';
import { HTMLFieldProps, connectField, filterDOMProps } from 'uniforms';
import { getValue } from '../../../utils';
import { CustomAutoField } from '../CustomAutoField';
import './CustomNestField.scss';

export type CustomNestFieldProps = HTMLFieldProps<
  object,
  HTMLDivElement,
  { properties?: Record<string, unknown>; helperText?: string; itemProps?: object }
>;

export const CustomNestField = connectField(
  ({
    children,
    error,
    errorMessage,
    fields,
    itemProps,
    label,
    name,
    showInlineError,
    disabled,
    ...props
  }: CustomNestFieldProps) => {
    const propertiesArray = Object.entries(props.properties ?? {}).reduce(
      (acc, [name, definition]) => {
        let labels: string = getValue(definition, 'labels', '');
        if (labels === '' || labels.includes('common')) {
          acc.common.push(name);
          return acc;
        }

        labels = labels.split(',').join(' ');

        acc.groups[labels] ??= [];
        acc.groups[labels].push(name);
        return acc;
      },
      { common: [], groups: {} } as { common: string[]; groups: Record<string, string[]> },
    );

    return (
      <Card data-testid={'nest-field'} {...filterDOMProps(props)} isExpanded>
        <CardHeader>
          <CardTitle>{label}</CardTitle>
        </CardHeader>
        <CardBody className="custom-nestfield-spacing">
          {propertiesArray.common.map((field) => (
            <CustomAutoField key={field} name={field} />
          ))}
        </CardBody>

        {Object.entries(propertiesArray.groups).map(([groupName, groupFields]) => (
          <ExpandableSection toggleText={`${groupName} properties`}>
            <CardBody className="custom-nestfield-spacing">
              {groupFields.map((field) => (
                <CustomAutoField key={field} name={field} />
              ))}
            </CardBody>
          </ExpandableSection>
        ))}
      </Card>
    );
  },
);
```

The difference here is that we iterate over the labels to create the groups dynamically, this way, for amqp we show other groups:
![image](https://github.com/KaotoIO/kaoto-next/assets/16512618/ff2ce13f-ecf4-4e38-9e5c-3e4168ace416)

It's not a final idea, since we would need to decorate the expandable titles with sentence case, and maybe there are other rough edges.

Reply:
Seems like there are quite many groups, but this is I suppose only the case for handful of connectors with large number of config properties, so even though for AMQP this looks like too many config dropdowns, this probably would be OK for most of the connectors.

The one thing that comes to my mind is how would we deduplicate the properties in the config, as in this solution one property could appear in more than one dropdown if I'm not mistaking.

Reply:
Nope, each parameter can be only in one group which is determined by the catalog attribute "group".

Reply:
Got it, I think it's possible to adjust it to match that, I'm just curious to know @tplevko opinion on the approach :smiley:

Reply:
hmm...actually I am not sure label is a good idea. that sounds more like tags